magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+                f"Old interface name: {network_interface} "
+                f"New interface name: eth{network_interface_index}."
+            )
+            cloud_init_content["network"]["ethernets"][
+                f"eth{network_interface_index}"
+            ] = cloud_init_content["network"]["ethernets"][network_interface]
+            del cloud_init_content["network"]["ethernets"][network_interface]
+            cloud_init_content["network"]["ethernets"][f"eth{network_interface_index}"][
+                "set-name"
+            ] = f"eth{network_interface_index}"
+
+        with open(cloud_init_config_file, "w") as cloud_init_file:
+            yaml.dump(cloud_init_content, cloud_init_file)
+
+        self.REBOOT_NEEDED = True
+
+    def _update_grub_cfg(self):
+        """Updates /boot/grub/grub.cfg."""
+        logger.info("Updating /boot/grub/grub.cfg...")
+        check_call(["grub-mkconfig", "-o", "/boot/grub/grub.cfg"])
+        self.REBOOT_NEEDED = True
+
+    @property
+    def _dns_configured(self) -> bool:
+        """Checks whether DNS has already been configured."""
+        with open("/etc/systemd/resolved.conf", "r") as dns_config:
+            for line in dns_config.readlines():
+                if f"DNS={self.DNS_ADDRESSES}" in line:
+                    return True
+        return False
+
+    def _configure_dns(self):
+        """Configures Ubuntu's DNS."""
+        logger.info("Configuring DNS...")
+        try:
+            os.symlink("/var/run/systemd/resolve/resolv.conf", "/etc/resolv.conf")
+        except FileExistsError:
+            os.remove("/etc/resolv.conf")
+            os.symlink("/var/run/systemd/resolve/resolv.conf", "/etc/resolv.conf")
+        original_resolved_conf_file = open("/etc/systemd/resolved.conf", "r")
+        original_resolved_conf_file_content = original_resolved_conf_file.readlines()
+        updated_resolved_conf_file_content = [
+            line.replace(line, f"DNS={self.DNS_ADDRESSES}\n")
+            if line.startswith("#DNS=") or line.startswith("DNS=")
+            else line
+            for line in original_resolved_conf_file_content
+        ]
+        original_resolved_conf_file.close()
+
+        updated_resolved_conf_file = open("/etc/systemd/resolved.conf", "w")
+        updated_resolved_conf_file.writelines(updated_resolved_conf_file_content)
+        updated_resolved_conf_file.close()
+        logger.info("Restarting systemd-resolved service...")
+        check_call(["service", "systemd-resolved", "restart"])
+
+        self.REBOOT_NEEDED = True
+
+    @property
+    def _network_interfaces_config_files_exist(self) -> bool:
+        """Checks whether configuration files for eth0 and eth1 exist in {self.INTERFACES_DIR}."""
+        if os.path.exists(f"{self.INTERFACES_DIR}/eth0") and os.path.exists(
+            f"{self.INTERFACES_DIR}/eth1"
+        ):
+            return True
+        else:
+            return False
```

Comment:
I think this might be a bit cleaner. What do you think?
```suggestion
       return bool(os.path.exists(f"{self.INTERFACES_DIR}/eth0") and os.path.exists(
            f"{self.INTERFACES_DIR}/eth1"
        ))
```

Reply:
I like that.
Changed.